bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/udp/UdpSenderService.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.revogi.internal.udp;
+
+import static java.util.stream.Collectors.toList;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.net.NetUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link UdpSenderService} is responsible for sending and receiving udp packets
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class UdpSenderService {
+
+    /**
+     * Limit timeout waiting time, as we have to deal with UDP
+     *
+     * How it works: for every loop, we'll wait a bit longer, so the timeout counter is multiplied with the timeout base
+     * value. Let max timeout count be 2 and timeout base value 800, then we'll have a maximum of loops of 3, waiting
+     * 800ms in the 1st loop, 1600ms in the 2nd loop and 2400ms in the third loop.
+     */
+    private static final int MAX_TIMEOUT_COUNT = 2;
+    public static final long TIMEOUT_BASE_VALUE = 800L;
+    private static final int REVOGI_PORT = 8888;
+
+    private final Logger logger = LoggerFactory.getLogger(UdpSenderService.class);
+    private final DatagramSocketWrapper datagramSocketWrapper;
+
+    public UdpSenderService(DatagramSocketWrapper datagramSocketWrapper) {
+        this.datagramSocketWrapper = datagramSocketWrapper;
+    }
+
+    public CompletableFuture<List<UdpResponse>> broadcastUpdDatagram(String content) {
+        List<String> allBroadcastAddresses = NetUtil.getAllBroadcastAddresses();
+        CompletableFuture<List<UdpResponse>> future = new CompletableFuture<>();
+        Executors.newCachedThreadPool().submit(() -> future.complete(allBroadcastAddresses.stream().map(address -> {
```

Comment:
Can you use the `ScheduledExecutorService` of the ThingHandler or the Thread of the `DiscoveryService`? I know this could result in quite extensive refactoring, but bindings should only create Threads if they really need to. E.g. listening on a socket permanently. Same for below.